File: java/Ocean.java
import java.util.ArrayList;
import java.util.List;
import java.lang.IllegalArgumentException;

public class Ocean{

    final static int WIDTH = 10;
    final static int HEIGHT = 10;
    private List<ArrayList<Square>> board;
    private List<Ship> ships;

    public Ocean(){
        board = new ArrayList<ArrayList<Square>>();
        ships = new ArrayList<>();

        ArrayList<Square> row;
        for (int i = 0; i < HEIGHT; i++){
            row = new ArrayList<>();
            for (int j = 0; j < WIDTH; j++){
                row.add(new Square());
            }
            board.add(row);
        }
    }

    public boolean addShip(int shipLength, int x, int y) throws IllegalArgumentException { 
        if (x < 0 || x >= WIDTH)
            throw new IllegalArgumentException("x coordinate should be in range 0..9");
        if (y < 0 || y >= HEIGHT)
            throw new IllegalArgumentException("y coordinate should be in range 0..9");
        if (shipLength < 1 || shipLength > 4)
            throw new IllegalArgumentException();
        
        List<Square> squares = new ArrayList<Square>();
        for (int i = 0 ; i < shipLength ; i++)
            squares.add(board.get(y).get(x + i));
        
        ships.add(new Ship(squares));
        System.out.println(ships);
        return true;
    }

    public boolean shoot(int x, int y) throws IllegalArgumentException {
        if (x < 0 || x >= WIDTH)
            throw new IllegalArgumentException("x coordinate should be in range 0..9");
        if (y < 0 || y >= HEIGHT)
            throw new IllegalArgumentException("y coordinate should be in range 0..9");

        Square square = board.get(x).get(y);
        
        square.shoot();

        for (Ship ship : ships) {

            if (ship.contains(square)) {
                square.setSymbol("X");
                return true;
            }
        }
        square.setSymbol("O");
        return false;

    }
    
    public String toString(){

        String str = " 12345678910\n";
        char alpha = 97;
        for (ArrayList<Square> row : board) {
            str = str + alpha;
            for (Square square : row) {
                str = str + square.toString();
            }
            str  = str + "\n";
            alpha++;
        }
        return str;
    }
}

File: csharp/Ocean.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace battleship_warmup_csharp
{
    public class Ocean{

        public static readonly int Width = 10;
        public static readonly int Height = 10;
        private readonly List<List<Square>> _board;
        private readonly List<Ship> _ships;

        public Ocean(){
            _board = new List<List<Square>>();
            _ships = new List<Ship>();

            for (var i = 0; i < Height; i++){
                var row = new List<Square>();
                for (var j = 0; j < Width; j++){
                    row.Add(new Square());
                }
                _board.Add(row);
            }
        }

        public bool AddShip(int shipLength, int x, int y) { 
            if (x < 0 || x >= Width)
                throw new ArgumentException("x coordinate should be in range 0..9");
            if (y < 0 || y >= Height)
                throw new ArgumentException("y coordinate should be in range 0..9");
            if (shipLength < 1 || shipLength > 4)
                throw new ArgumentException();
            
            var squares = new List<Square>();
            for (var i = 0 ; i < shipLength ; i++)
                squares.Add(_board[y][x + i]);
            
            _ships.Add(new Ship(squares));
            Console.WriteLine(_ships);
            return true;
        }

        public bool Shoot(int x, int y) {
            if (x < 0 || x >= Width)
                throw new ArgumentException("x coordinate should be in range 0..9");
            if (y < 0 || y >= Height)
                throw new ArgumentException("y coordinate should be in range 0..9");

            var square = _board[x][y];
            
            square.Shoot();

            if (_ships.Any(ship => ship.Contains(square)))
            {
                square.Symbol = "X";
                return true;
            }

            square.Symbol = "O";
            return false;

        }
        
        public override string ToString(){

            var str = " 12345678910\n";
            var ascii = 97;
            foreach (var row in _board) {
                str += (char)ascii;
                str = row.Aggregate(str, (current, square) => current + square);
                str += "\n";
                ascii++;
            }
            return str;
        }
    }
}